voxblox/include/voxblox/integrator/tsdf_integrator.h

```diff
@@ -406,23 +492,24 @@ class MergedTsdfIntegrator : public TsdfIntegratorBase {
         }
       }
 
-      voxel_info.block_idx = getGridIndexFromPoint(
-          global_voxel_idx.cast<FloatingPoint>(), voxels_per_side_inv_);
-
-      voxel_info.local_voxel_idx =
-          getLocalFromGlobalVoxelIndex(global_voxel_idx, voxels_per_side_);
+      TsdfVoxel* voxel = getVoxel(global_voxel_idx, &block, &last_block_idx);
+      if (voxel != nullptr) {
+        const Point voxel_center_G =
+            getCenterPointFromGridIndex(global_voxel_idx, voxel_size_);
 
-      voxel_update_queue->push(voxel_info);
+        updateTsdfVoxel(origin, merged_point_G, voxel_center_G, merged_color,
+                        config_.default_truncation_distance, merged_weight,
+                        voxel);
+      }
     }
-    update_voxels_timer.Stop();
   }
 
   void integrateVoxels(
       const Transformation& T_G_C, const Pointcloud& points_C,
       const Colors& colors, bool enable_anti_grazing, bool clearing_ray,
       const BlockHashMapType<std::vector<size_t>>::type& voxel_map,
       const BlockHashMapType<std::vector<size_t>>::type& clear_map,
-      std::queue<VoxelInfo>* voxel_update_queue, size_t tid) {
+      size_t tid) {
```

Comment:
the const for fundamental types might not be strictly necessary because the params are a copy anyways, so no harm is done by changing them. However having them const would raise a compilation error should you accidentally assign something to them, which helps prevent bugs. :) (see [here, second answer](https://stackoverflow.com/questions/117293/use-of-const-for-function-parameters))